packages/react/src/hooks/useTranscription.ts

```diff
@@ -0,0 +1,25 @@
+import type { RemoteParticipant, RemoteTrackPublication, Transcription } from 'livekit-client';
+import { createTranscriptionObserver } from '@livekit/components-core';
+import * as React from 'react';
+import { useRoomContext } from '../context';
+
+/**
+ * @alpha
+ */
+export function useTranscription(
+  onTranscription: (
```

Comment:
Does it make sense to wrap this? instead of having the user directly use the JS SDK event?

I'm thinking that might be more direct and requires less translation. Ideally the interface we offer in components contains the state of the last N transcriptions? like `useTranscriptions() -> []Transcription`, and similarly for participants: `useParticipantTranscriptions(p)`

and Transcription could be:

```
interface Transcription {
  participant,
  publication,
  segments,
```

Reply:
this is just thinking out loud.. maybe each transcription should contain a single segment? so we can properly sort it by time. especially when metadata sync is in

Reply:
yeah, tagging you for this PR was a bit premature, sorry about that. 
I'll play around with some designs on how to handle transcription state in conjunction with track syncs.